### PR TITLE
Fix MM-42096

### PR DESCRIPTION
**Pull request by hanzei**
#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-42096

### CODE REVIEW

**Review by levb — APPROVED** [review]

File: server/plugin/webhook.go [hunk 1]
```diff
@@ -147,13 +147,25 @@ func (wb *WebhookBroker) Close() {
 func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
 	config := p.getConfiguration()
 
-	signature := r.Header.Get("X-Hub-Signature")
-	body, err := ioutil.ReadAll(r.Body)
+	body, err := io.ReadAll(r.Body)
```
Comment (jupenur) [inline server/plugin/webhook.go:150]: I'm not sure what's happening here; why is this any better than the original? We're still reading the whole body regardless of size, right?

Comment (hanzei) [reply server/plugin/webhook.go:150]: Yes, this PR doesn't change how the body is read, But it changes how the body is parsed. We now check the HMAC before calling `github.ParseWebHook`.

Comment (jupenur) [reply server/plugin/webhook.go:150]: Got it; and I think that's fine and a good defense-in-depth improvement, knowing that there is a separate size limit in the server.

**Review by jupenur — APPROVED** [review]
LGTM

**Pull request merged**